backend-lib/src/main/rust/lib.rs

```diff
@@ -2540,7 +2526,7 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_model_TorClient_createTor
     tor_dir: JString<'local>,
 ) -> jlong {
     let res = catch_unwind(&mut env, |env| {
-        let tor_dir = utils::java_string_to_rust(env, &tor_dir);
+        let tor_dir = utils::java_string_to_rust(env, &tor_dir)?;
         let tor_dir = Path::new(&tor_dir);
```

Comment:
It happens that all of the uses of `Path::new` in this file (7 of them) are from Java strings. So this helper:
```rust
fn path_from_jni(env: &JNIEnv, path_str: &JString) -> anyhow::Result<Path> {
    Ok(Path::new(utils::java_string_to_rust(env, path_str)?))
}
```
would let all paths be immediately typed as `Path` and eliminate all of the `Path::new`s.

Reply:
We cannot use `Path::new` in the helper like this because it is a reference type, and this would lead to an early drop of a temporary variable. We are instead forced to reallocate in a `PathBuf`. But that's fine.